src/main/java/io/cdap/plugin/http/sink/batch/HTTPSinkConfig.java

```diff
@@ -0,0 +1,436 @@
+    builder.setFollowRedirects(copy.getFollowRedirects());
+    builder.setDisableSSLValidation(copy.getDisableSSLValidation());
+    builder.setNumRetries(copy.getNumRetries());
+    builder.setConnectTimeout(copy.getConnectTimeout());
+    builder.setReadTimeout(copy.getReadTimeout());
+    builder.setFailOnNon200Response(copy.getFailOnNon200Response());
+    return builder;
+  }
+
+  public String getUrl() {
+    return url;
+  }
+
+  public String getMethod() {
+    return method;
+  }
+
+  public Integer getBatchSize() {
+    return batchSize;
+  }
+
+  @Nullable
+  public String getDelimiterForMessages() {
+    return delimiterForMessages;
+  }
+
+  public String getMessageFormat() {
+    return messageFormat;
+  }
+
+  @Nullable
+  public String getBody() {
+    return body;
+  }
+
+  @Nullable
+  public String getRequestHeaders() {
+    return requestHeaders;
+  }
+
+  public String getCharset() {
+    return charset;
+  }
+
+  public Boolean getFollowRedirects() {
+    return followRedirects;
+  }
+
+  public Boolean getDisableSSLValidation() {
+    return disableSSLValidation;
+  }
+
+  public Integer getNumRetries() {
+    return numRetries;
+  }
+
+  @Nullable
+  public Integer getConnectTimeout() {
+    return connectTimeout;
+  }
+
+  @Nullable
+  public Integer getReadTimeout() {
+    return readTimeout;
+  }
+
+  public Boolean getFailOnNon200Response() {
+    return failOnNon200Response;
+  }
+
+  public Map<String, String> getRequestHeadersMap() {
+    return convertHeadersToMap(requestHeaders);
+  }
+
+  public void validate(FailureCollector collector) {
+    try {
+      new URL(url);
+    } catch (MalformedURLException e) {
+      collector.addFailure(String.format("URL '%s' is malformed: %s", url, e.getMessage()), null)
+        .withConfigProperty(URL);
+    }
+
+    if (!containsMacro(CONNECTION_TIMEOUT) && connectTimeout < 0) {
+      collector.addFailure(
+        String.format("Invalid connectTimeout %d.", connectTimeout), "Timeout must be 0 or a positive number.")
+        .withConfigProperty(CONNECTION_TIMEOUT);
+    }
+
+    try {
+      convertHeadersToMap(requestHeaders);
+    } catch (IllegalArgumentException e) {
+      collector.addFailure(e.getMessage(), null)
+        .withConfigProperty(REQUEST_HEADERS);
+    }
+
+    if (!containsMacro(METHOD) && !METHODS.contains(method.toUpperCase())) {
+      collector.addFailure(
+        String.format("Invalid request method %s, must be one of %s.", method, Joiner.on(',').join(METHODS)), null)
+        .withConfigProperty(METHOD);
+    }
+
+    if (!containsMacro(NUM_RETRIES) && numRetries < 0) {
+      collector.addFailure(
+        String.format("Invalid numRetries %d.", numRetries), "Retries cannot be a negative number.")
+        .withConfigProperty(NUM_RETRIES);
+    }
+
+    if (!containsMacro(READ_TIMEOUT) && readTimeout < 0) {
+      collector.addFailure(
+        String.format("Invalid readTimeout %d. ", readTimeout), "Timeout must be 0 or a positive number.")
```

Comment:
to be consistent with the retry message, 'Timeout cannot be a negative number.'

Reply:
use the label shown in the UI instead of the variable name readTimeout

Reply:
done